lib/couchdb/index.js

```diff
@@ -83,6 +83,29 @@ const upsert = R.curry((couch, bucket, keys, doc) =>
 )(doc)))
 
 
+const bulk_upsert = R.curry((couch, bucket, keys, docs) =>
+  Bluebird.map(docs, (doc) =>
+    Bluebird.resolve(keys)
+    .then(R.flip(R.pick)(doc))
+    .then(R.ifElse(
```

Comment:
looks like the H&M for this anonymous function will be 

`anon :: Predicates -> doc | [ doc ]`

Then the next anon function on line 99 will take the head of the passed value and couchify.

I think R.head might run on a single document that's returned in the above's `R.always(doc)` which would be bad right?